protocols/decentralizedbaiprotocol.py

```diff
@@ -0,0 +1,122 @@
+from random import choice
+
+
+from absl import flags
+from absl import logging
+
+import numpy as np
+
+from .utils import Protocol
+
+FLAGS = flags.FLAGS
+
+__all__ = ['DecentralizedBAIProtocol']
+
+
+class DecentralizedBAIProtocol(Protocol):
+  """Decentralized Best Arm Identification Protocol
+  """
+
+  def __init__(self, pars):
+    self.__messages = []
+    self.__num_players = pars['num_players']
+
+  @property
+  def type(self):
+    return 'DecentralizedBAIProtocol'
+
+  @property
+  def _num_players(self):
+    return self.__num_players
+
+  @property
+  def _budget(self):
+    return self.__budget
+
+  @property
+  def _fail_prob(self):
+    return self.__fail_prob
+
+  def _one_round(self):
+    # sample player
+    k = choice(self.__playable_players)
+    player = self._players[k]
+    bandit = self._bandits[k]
+
+    # play player and broadcast message
+    action = player.learner_choice(self.__messages)
+    if action == -1:
+      return k
+    feedback = bandit.feed(action)
+    player.update(action, feedback)
+
+    message = player._broadcast_message(action, feedback)
```

Comment:
Can you change to `broadcast_message` since `_` prefix is intended for no outside usage?